.github/workflows/cloud-deploy-reusable-app-service-deploy.yml

```diff
@@ -0,0 +1,217 @@
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      service_name:
+        required: true
+        type: string
+      docker_image:
+        required: true
+        type: string
+      sentry_project:
+        required: false
+        type: string
+      New_Relic_GUID:
+        required: false
+        type: string
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  infrastructure_data:
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    permissions:
+      contents: read
+      deployments: write
+    env:
+      TF_WORKSPACE: ${{ vars.TF_WORKSPACE }}
+    outputs:
+      ecs_container_name: ${{ steps.terraform.outputs.ecs_container_name }}
+      ecs_service: ${{ steps.terraform.outputs.ecs_service }}
+      ecs_cluster: ${{ steps.terraform.outputs.ecs_cluster }}
+      task_name: ${{ steps.terraform.outputs.task_name }}
+      aws_region: ${{ steps.terraform.outputs.aws_region }}
+      services_to_deploy: ${{ steps.terraform.outputs.queue_workers_services }}
+    steps:
+      - run: echo "Deploying ${{ inputs.service_name }} to ${{ vars.TF_WORKSPACE }} And Docker Tag ${{ inputs.docker_image }}"
+
+      - name: Checkout cloud infra
+        uses: actions/checkout@v4
+        with:
+          repository: novuhq/cloud-infra
+          token: ${{ secrets.GH_PACKAGES }}
+          path: cloud-infra
+
+      - name: Terraform setup
+        uses: hashicorp/setup-terraform@v3
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_version: 1.5.5
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        working-directory: cloud-infra/terraform/novu/aws
+        run: terraform init
+
+      - name: Replace dash with underscore in service_name
+        id: replace_dash
+        run: echo "service_name_under=$(echo '${{ inputs.service_name }}' | tr '-' '_')" >> $GITHUB_ENV
+
+      - name: Terraform get output
+        working-directory: cloud-infra/terraform/novu/aws
+        id: terraform
+        env:
+          SERVICE_NAME: ${{ env.service_name_under }}
+        run: |
+          echo "ecs_container_name=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_container_name | jq -r .)" >> $GITHUB_OUTPUT
+          echo "ecs_service=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_service | jq -r .)" >> $GITHUB_OUTPUT
+          echo "ecs_cluster=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_cluster | jq -r .)" >> $GITHUB_OUTPUT
+          echo "task_name=$(terraform output -json ${{ env.SERVICE_NAME }}_task_name | jq -r .)" >> $GITHUB_OUTPUT
+          echo "aws_region=$(terraform output -json aws_region | jq -r .)" >> $GITHUB_OUTPUT
+          echo "queue_workers_services=$(terraform output -json queue_workers_services)" >> $GITHUB_OUTPUT
+
+  deploy_service_worker:
+    if: inputs.service_name == 'worker'
```

Comment:
@AliaksandrRyzhou Please add a comment why workers need to be treated separately for now. Actually, is this still the case or can we remove this conditional since we fetch the latest ECS definition from AW?S

Reply:
If we take a look a difference is not too big here. 
![Screenshot from 2024-07-26 15-52-04](https://github.com/user-attachments/assets/0671dc2d-5a19-4334-a963-5ff5e0d0d120)

The worker service only has a few services inside of ECS cluster and so that to get a task definition for each of them and deploy each of them we have to receive all these names to build a matrix by using the terraform the output values of queue_workers_services. We get the services, build matrix for each of them and update the TD for each of them. So that we can avoid using a two jobs we just need to create a terraform output for each service and leave the job with matrix based on a queue of each service, in any another case we use conditions somewher in GitHub Actions.